src/idl_parser.cpp

```diff
@@ -1314,6 +1350,38 @@ CheckedError Parser::ParseSingleValue(Value &e) {
   return NoError();
 }
 
+void Parser::InitializeNestedFlatbufferParser(Parser& nestedParser, Value *const nested) {
+  assert(nested);
```

Comment:
I don't think all of these need to be copied for JSON parsing to work, probably only `enums_`.